pkg/webhook/mutation/pod/init_container_test.go

```diff
@@ -238,4 +239,35 @@ func TestInitContainerResources(t *testing.T) {
 
 		require.Empty(t, initResources)
 	})
+
+	t.Run("should have default if metadata enrichment is enabled", func(t *testing.T) {
+		dk := getTestDynakubeDefaultAppMon()
+		dk.Spec.MetadataEnrichment.Enabled = ptr.To(true)
+
+		initResources := initContainerResources(*dk)
+
+		assert.Equal(t, defaultInitContainerResources(), initResources)
+	})
+	t.Run("should have default if metadata enrichment is enabled and csi is disabled", func(t *testing.T) {
+		installconfig.SetModulesOverride(t, installconfig.Modules{CSIDriver: false})
+
+		dk := getTestDynakubeDefaultAppMon()
+		dk.Spec.MetadataEnrichment.Enabled = ptr.To(true)
+
+		initResources := initContainerResources(*dk)
+
+		assert.Equal(t, defaultInitContainerResources(), initResources)
+	})
+	t.Run("should have no limit if OA is enabled and csi is disabled", func(t *testing.T) {
+		installconfig.SetModulesOverride(t, installconfig.Modules{CSIDriver: false})
+
+		dk := getTestDynakubeDefaultAppMon()
+
+		dk.Spec.MetadataEnrichment.Enabled = ptr.To(true)
+		dk.Spec.OneAgent.CloudNativeFullStack = &oneagent.CloudNativeFullStackSpec{}
```

Comment:
kinda my mistake, but this is not a realistic scenario, as no-csi + cloudnative is not possible, it should be appmon

Reply:
of course - I should've question this. But I get it now and understood it. Thanks!

Reply:
fixed unit test https://github.com/Dynatrace/dynatrace-operator/pull/4288/commits/1d37ee6e3c6c51fff3e8f6e41a2da3e53aec6012